donut/modules/voting/ranked_pairs.py

```diff
@@ -0,0 +1,50 @@
+from itertools import product
+
+WINNERS_TO_LIST = 3
+
+
+def winners(responses):
```

Comment:
very nice. as far as i can tell seems correct, though algorithms were never my strong point haha. make sure you include everything they talk about in the bylaws. for one, and i'm unsure about this, is the ranking you define here mathematically equivalent to the ranking specified in the bylaws (with respect to picking multiple winners) where you choose the one winner, remove all votes relevant to that winner, and rerun the algorithm to generate the next winner? i am not sure if it is.
Also, it is worth mentioning that some positions require more than one winner, in which case we shouldn't display the results as rankings but rather as winners, or at least 2 first places. lol.

Reply:
I'm pretty sure it is equivalent. Wikipedia says that you can repeatedly remove the winner and re-run the algorithm, but it also shows that you can just build the optimal DAG once and then a topological sort on it gives you the ranking.

Reply:
Doesn't seem so critical to me whether it says "1. Candidate A; 2. Candidate B; 3. Candidate C" or "Top 3: A, B, and C". I think it is unnecessary extra logic to store how many winners each elected position question has.